File: static_backup/scan_languages.py
import os
import re

en_dir = r"f:\موقع شركة التسويق الطبيه\stitch_digital_health_agency_ui_ux\website\en"
ar_dir = r"f:\موقع شركة التسويق الطبيه\stitch_digital_health_agency_ui_ux\website"

arabic_regex = re.compile(r'[\u0600-\u06FF\u0750-\u077F\u08A0-\u08FF\uFB50-\uFDFF\uFE70-\uFEFF]')

print("--- SCANNING ENGLISH PAGES FOR ARABIC CHARACTERS ---")
for file in os.listdir(en_dir):
    if file.endswith('.html'):
        path = os.path.join(en_dir, file)
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Remove script tags and style tags content to avoid matching non-displayed text
        clean_content = re.sub(r'<script.*?>.*?</script>', '', content, flags=re.DOTALL)
        clean_content = re.sub(r'<style.*?>.*?</style>', '', clean_content, flags=re.DOTALL)
        
        matches = arabic_regex.findall(clean_content)
        if matches:
            print(f"Found {len(matches)} Arabic characters in English page: {file}")
            # print first 5 matches or sample contexts
            for m in re.finditer(arabic_regex, clean_content):
                start = max(0, m.start() - 30)
                end = min(len(clean_content), m.end() + 30)
                context = clean_content[start:end].replace('\n', ' ')
                print(f"  Context: ... {context} ...")
                break

print("\n--- SCANNING ARABIC PAGES FOR ENGLISH WORDS IN DISPLAYED TEXT ---")
for file in os.listdir(ar_dir):
    if file.endswith('.html'):
        path = os.path.join(ar_dir, file)
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # We want to check for English words inside elements (e.g. not inside tags, script, style)
        # Let's strip script, style, comments, and tags
        clean_content = re.sub(r'<!--.*?-->', '', content, flags=re.DOTALL)
        clean_content = re.sub(r'<script.*?>.*?</script>', '', clean_content, flags=re.DOTALL)
        clean_content = re.sub(r'<style.*?>.*?</style>', '', clean_content, flags=re.DOTALL)
        
        # Now remove tags but keep their inner text (by replacing tags with space)
        text_only = re.sub(r'<[^>]+>', ' ', clean_content)
        
        # Find any English words (sequences of a-zA-Z)
        # Filter out common things like Google Font names, etc. that might appear, but let's check
        eng_words = re.findall(r'\b[a-zA-Z]{2,}\b', text_only)
        # Let's clean up words that are actually inside text and not HTML entity leftovers like amp, gt, lt, etc.
        words_to_show = []
        for w in eng_words:
            if w.lower() not in ['amp', 'gt', 'lt', 'quot', 'nbsp']:
                words_to_show.append(w)
        if words_to_show:
            unique_eng = set(words_to_show)
            print(f"Found {len(words_to_show)} English words (unique: {unique_eng}) in Arabic page: {file}")

File: static_backup/scan_languages.js
const fs = require('fs');
const path = require('path');

const enDir = 'f:\\موقع شركة التسويق الطبيه\\stitch_digital_health_agency_ui_ux\\website\\en';
const arDir = 'f:\\موقع شركة التسويق الطبيه\\stitch_digital_health_agency_ui_ux\\website';

const arabicRegex = /[\u0600-\u06FF\u0750-\u077F\u08A0-\u08FF\uFB50-\uFDFF\uFE70-\uFEFF]/g;

console.log('--- SCANNING ENGLISH PAGES FOR ARABIC CHARACTERS ---');
try {
  const enFiles = fs.readdirSync(enDir);
  enFiles.forEach(file => {
    if (file.endsWith('.html')) {
      const filePath = path.join(enDir, file);
      const content = fs.readFileSync(filePath, 'utf8');
      
      // Strip script and style
      let cleanContent = content.replace(/<script[\s\S]*?>[\s\S]*?<\/script>/gi, '');
      cleanContent = cleanContent.replace(/<style[\s\S]*?>[\s\S]*?<\/style>/gi, '');
      
      const matches = cleanContent.match(arabicRegex);
      if (matches) {
        console.log(`Found ${matches.length} Arabic characters in English page: ${file}`);
        // Find context
        const firstMatch = arabicRegex.exec(cleanContent);
        if (firstMatch) {
          const idx = firstMatch.index;
          const start = Math.max(0, idx - 30);
          const end = Math.min(cleanContent.length, idx + 30);
          const context = cleanContent.slice(start, end).replace(/\n/g, ' ');
          console.log(`  Context: ... ${context} ...`);
        }
      }
    }
  });
} catch (e) {
  console.error('Error scanning English files:', e.message);
}

console.log('\n--- SCANNING ARABIC PAGES FOR ENGLISH WORDS IN DISPLAYED TEXT ---');
try {
  const arFiles = fs.readdirSync(arDir);
  arFiles.forEach(file => {
    if (file.endsWith('.html')) {
      const filePath = path.join(arDir, file);
      const content = fs.readFileSync(filePath, 'utf8');
      
      // Strip comments, script, style
      let cleanContent = content.replace(/<!--[\s\S]*?-->/g, '');
      cleanContent = cleanContent.replace(/<script[\s\S]*?>[\s\S]*?<\/script>/gi, '');
      cleanContent = cleanContent.replace(/<style[\s\S]*?>[\s\S]*?<\/style>/gi, '');
      
      // Strip HTML tags
      const textOnly = cleanContent.replace(/<[^>]+>/g, ' ');
      
      // Find English words [a-zA-Z]{2,}
      const engWords = textOnly.match(/\b[a-zA-Z]{2,}\b/g) || [];
      const filtered = engWords.filter(w => !['amp', 'gt', 'lt', 'quot', 'nbsp'].includes(w.toLowerCase()));
      
      if (filtered.length > 0) {
        const unique = Array.from(new Set(filtered));
        console.log(`Found ${filtered.length} English words (unique: ${unique.join(', ')}) in Arabic page: ${file}`);
      }
    }
  });
} catch (e) {
  console.error('Error scanning Arabic files:', e.message);
}
